packages/react/src/components/SplitButton/SplitButton.tsx

```diff
@@ -0,0 +1,204 @@
+    onMenuItemClick: PropTypes.func,
+    open: PropTypes.bool,
+    primary: customPropTypes.every([customPropTypes.disallow(['secondary']), PropTypes.bool]),
+    secondary: customPropTypes.every([customPropTypes.disallow(['primary']), PropTypes.bool]),
+    toggleButton: customPropTypes.itemShorthand,
+  }
+
+  static defaultProps = {
+    accessibility: splitButtonBehavior,
+    as: 'span',
+  }
+
+  static autoControlledProps = ['open']
+
+  getInitialAutoControlledState(): SplitButtonState {
+    return {
+      isFromKeyboard: false,
+      open: false,
+    }
+  }
+
+  renderComponent({
+    ElementType,
+    classes,
+    accessibility,
+    styles,
+    unhandledProps,
+  }): React.ReactNode {
+    const { button, menu, primary, secondary, toggleButton } = this.props
+
+    return (
+      <ElementType className={classes.root} {...accessibility.attributes.root} {...unhandledProps}>
+        {MenuButton.create(
+          {},
+          {
+            defaultProps: {
+              accessibility: accessibility.childBehaviors.menuButton,
+              menu,
+              styles: styles.button,
+              on: [],
+              open: this.state.open,
+              onOpenChange: (e: React.SyntheticEvent, { open }: PopupProps) => {
+                e.stopPropagation()
+                this.setState({ open })
+              },
+              trigger: Button.create(button, {
+                defaultProps: {
+                  styles: styles.button,
+                  primary,
+                  secondary,
+                },
+              }),
+            },
+            overrideProps: (predefinedProps: MenuButtonProps) => ({
+              onMenuItemClick: (e: React.SyntheticEvent, menuItemProps: MenuItemProps) => {
+                this.setState({ open: false })
+
+                _.invoke(predefinedProps, 'onMenuItemClick', e, menuItemProps)
+                _.invoke(this.props, 'onMenuItemClick', e, menuItemProps)
+              },
+              onClick: (e: React.SyntheticEvent, buttonProps: ButtonProps) => {
+                _.invoke(predefinedProps, 'onClick', e, buttonProps)
+                _.invoke(this.props, 'onClick', e, buttonProps)
+              },
+              onFocus: () => {
+                this.setState({ isFromKeyboard: isFromKeyboard() })
+              },
+            }),
+          },
+        )}
+        {Button.create(toggleButton || {}, {
+          defaultProps: {
+            className: SplitButton.slotClassNames.toggleButton,
+            styles: styles.secondaryButton,
+            icon: 'chevron-down',
+            iconOnly: true,
+            primary,
+            secondary,
+            ...accessibility.attributes.secondaryButton,
+          },
+          overrideProps: (predefinedProps: ButtonProps) => ({
+            ...(!this.state.open && {
+              onClick: (e: React.SyntheticEvent, buttonProps: ButtonProps) => {
+                this.setState(state => ({ open: !state.open }))
```

Comment:
```suggestion
                _.invoke(predefinedProps, 'onClick', e, buttonProps)
                this.setState(state => ({ open: !state.open }))
```

Original handler should invoked